### PR TITLE
add README in Chinese

### PR DESCRIPTION
**Pull request by raoenhui**
add README in Chinese

### CODE REVIEW

File: readme-cn.md [hunk 1]
```diff
@@ -0,0 +1,1799 @@
...
+  onDemandEntries: {
+    // period (in ms) where the server will keep pages in the buffer
+    maxInactiveAge: 25 * 1000,
+    // number of pages that should be kept simultaneously without being disposed
+    pagesBufferLength: 2,
+  }
+}
+```
+
+这个只是在开发环境才有的功能。如果你在生成环境中想缓存SSR页面，请查看[SSR-caching](https://github.com/zeit/next.js/tree/canary/examples/ssr-caching
```
Comment (quietshu) [inline readme-cn.md]: Missing a `)` at the end of the URL?

File: readme-cn.md [hunk 2]
```diff
@@ -0,0 +1,1799 @@
...
+})
+```
+
+`next`的API如下所示
+- `next(opts: object)`
+
+opts的属性如下:
+- `dev` (`boolean`) 判断 Next.js应用是否在开发环境 - 默认 `false`
+- `dir` (`string`) Next项目路径 - 默认 `'.'`
+- `quiet` (`boolean`) Hide error messages containing server information - 默认 `false`
```
Comment (quietshu) [inline readme-cn.md]: Not translated.

File: readme-cn.md [hunk 3]
```diff
@@ -0,0 +1,1799 @@
...
+// next.config.js
+module.exports = {
+  pageExtensions: ['jsx', 'js']
+}
+```
+
+<a id="configuring-the-build-id" style="display: none"></a>
+#### 配置构建ID
+
+Next.js使用一个常量来判断你的应用服务是哪个版本。因为多个服务部署应用，那`next build`命令将会在多个服务器中运行，为了保持同一个构建ID，可以配置`generateBuildId`函数：
```
Comment (quietshu) [inline readme-cn.md]: `因为多个服务部署应用，那 next build 命令将会在多个服务器中运行` was wrongly translated.

File: readme-cn.md [hunk 4]
```diff
@@ -0,0 +1,1799 @@
...
+### 自定义babel配置
+
+<p><details>
+  <summary><b>Examples</b></summary>
+  <ul><li><a href="./examples/with-custom-babel-config">Custom babel configuration</a></li></ul>
+</details></p>
+
+为了扩展方便我们使用`babel`，可以在应用根目录新建`.babelrc`文件，该文件可配置。
+
+如果有该文件，我们将会考虑来源的真实性，因此同样你需要定义next，也就是 `next/babel`预设。
```
Comment (quietshu) [inline readme-cn.md]: “source of truth” should be “数据源”, not “来源的真实性”.

File: readme-cn.md [hunk 5]
```diff
@@ -0,0 +1,1799 @@
...
+- [设置301重定向](https://www.raygesualdo.com/posts/301-redirects-with-nextjs/)
+- [只处理服务器端模块](https://arunoda.me/blog/ssr-and-server-only-modules)
+- [构建项目React-Material-UI-Next-Express-Mongoose-Mongodb](https://github.com/builderbook/builderbook)
+- [构建一个SaaS产品 React-Material-UI-Next-MobX-Express-Mongoose-MongoDB-TypeScript](https://github.com/async-labs/saas)
+
+<a id="faq" style="display: none"></a>
+## 问答
+
+<details>
+  <summary>这个产品准备好了吗？</summary>
```
Comment (quietshu) [inline readme-cn.md]: Production ready: “可以用于生产环境吗？”.

**Review by shuding — COMMENTED** [review]
Thanks for the translation! It's great to see the Chinese README 👍

However it seems that it's not finished yet? (as I commented and also in FAQ and some other sections)

Also, some terms such as "NPM", "JS" and "CSS" should not be "npm", "js" and "css". And the spaces between Chinese words and alphabets are kind of confused. i.e. `而不依靠 Node.js服务。` there's a whitespace before `Node.js` (which is very nice!) but we should also add a whitespace after it ;) It improves the readability of Chinese tech docs.

Minor thing: I think the file name (readme-cn.md) would be better as uppercase and use "zh-CN" (README-zh-CN.md) to follow README.md.

File: readme-cn.md (hunk 3 above)
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion，I re-translated it like "Next.js使用构建时生成的常量来标识你的应用服务是哪个版本。在每台服务器上运行构建命令时，可能会导致多服务器部署出现问题。为了保持同一个构建ID，可以配置`generateBuildId`函数："

File: readme-cn.md [hunk 6]
```diff
@@ -0,0 +1,1799 @@
+<img width="112" alt="screen shot 2016-10-25 at 2 37 27 pm" src="https://cloud.githubusercontent.com/assets/13041/19686250/971bf7f8-9ac0-11e6-975c-188defd82df1.png">
+
+[![NPM version](https://img.shields.io/npm/v/next.svg)](https://www.npmjs.com/package/next)
+[![Build Status](https://travis-ci.org/zeit/next.js.svg?branch=master)](https://travis-ci.org/zeit/next.js)
+[![Build status](https://ci.appveyor.com/api/projects/status/gqp5hs71l3ebtx1r/branch/master?svg=true)](https://ci.appveyor.com/project/arunoda/next-js/branch/master)
+[![Coverage Status](https://coveralls.io/repos/zeit/next.js/badge.svg?branch=master)](https://coveralls.io/r/zeit/next.js?branch=master)
+[![Join the community on Spectrum](https://withspectrum.github.io/badge/badge.svg)](https://spectrum.chat/next-js)
+
+Next.js 是一个轻巧的React服务端渲染应用框架。
```
[the remark this answers is not in the document]
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have modified it.

File: readme-cn.md [hunk 7]
```diff
@@ -0,0 +1,1799 @@
...
+
+相信你注意到，当页面渲染时加载数据，我们使用了一个异步方法`getInitialProps`。它能异步获取js普通对象，并绑定在`props`上
+
+当服务渲染时，`getInitialProps`将会把数据序列化，就像`JSON.stringify`。所以确保`getInitialProps`返回的是一个普通js对象，而不是`Date`, `Map` 或 `Set`类型。
+
+当页面初次加载时，`getInitialProps`只会在服务端执行一次。`getInitialProps`只有在路由切换的时候（如`Link`组件跳转或路由自定义跳转）时，客户端的才会被执行。
+
+当页面初始化加载时，`getInitialProps`只会加载在服务端。只有当路由跳转（`Link`组件跳转或API方法跳转）时，客户端才会执行`getInitialProps`。
+
+_Note: `getInitialProps` can **not** be used in children components. Only in `pages`._
```
[the remark this answers is not in the document]
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have translated it.

File: readme-cn.md [hunk 8]
```diff
@@ -0,0 +1,1799 @@
...
+      <meta name="viewport" content="initial-scale=1.2, width=device-width" key="viewport" />
+    </Head>
+    <p>Hello world!</p>
+  </div>
+)
+```
+
+只有第二个`<meta name="viewport" />`才被渲染。
+
+注意：在卸载组件时，' <head> '的内容将被清除。请确保每个页面都在其`<head>`定义了所需要的内容，而不是假设其他页面已经加过了
```
[the remark this answers is not in the document]
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have removed it.

File: readme-cn.md [hunk 9]
```diff
@@ -0,0 +1,1799 @@
...
+注意：`getInitialProps`将不能使用在子组件中。只能使用在`pages`页面中。
+
+<br/>
+
+> 只有服务端用到的模块放在 `getInitialProps`里，请确保正确的导入了它们，可参考[import them properly](https://arunoda.me/blog/ssr-and-server-only-modules)。
+> 否则会拖慢你的应用速度。
+
+<br/>
+
+你也可以给5⃣无状态组件定义 `getInitialProps`：
```
[the remark this answers is not in the document]
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have removed it.

File: readme-cn.md [hunk 10]
```diff
@@ -0,0 +1,1799 @@
...
+
+注意：可以使用[`<Link prefetch>`](#prefetching-pages)使链接和预加载在后台同时进行，来达到页面的最佳性能。
+
+客户端路由行为与浏览器很相似：
+
+1. 组件获取
+2. If it defines `getInitialProps`, data is fetched. If an error occurs, `_error.js` is rendered2. 
+2. 如果组件定义了`getInitialProps`，数据获取了。如果有错误情况将会渲染 `_error.js`。
+3. After 1 and 2 complete, `pushState` is performed and the new component is rendered
+3. 1和2都完成了，`pushState`执行，新组件被渲染。
```
[the remark this answers is not in the document]
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have translated it.

File: readme-cn.md (hunk 1 above)
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have fixed it.

File: readme-cn.md (hunk 2 above)
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have translated it.

File: readme-cn.md (hunk 4 above)
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have modified it.

File: readme-cn.md (hunk 5 above)
Comment (raoenhui) [reply readme-cn.md]: Thank you for your suggestion, I have modified it.

**Review by shuding — APPROVED** [review]
Thanks a lot @raoenhui 🙏 it looks perfect! 
@timneutkens can we merge this PR and add a link in README.md?

**Comment by rauchg** [comment]
@timneutkens I'm +1 on merging this, and then we can port it to nextjs.org once we implement i18n there.

**Comment by timneutkens** [comment]
Okay let's ship it then 👍

**Pull request merged**